astropy/io/ascii/ui.py

```diff
@@ -660,9 +660,27 @@ def write(table, output=None,  format=None, Writer=None, fast_writer=True, **kwa
         List of names to exclude from output (applied after ``include_names``)
     fast_writer : bool
         Whether to use the fast Cython writer (default= ``True``)
+    overwrite : bool
+        Whether to overwrite an existing file (default= ``False``).
+        This parameter is ignored when the output is not a string
+        (e.g., a file object).
+        Since the default behaviour is changed (previously, files would
+        always be overwritten), an AstropyDeprecationWarning is issued
+        when this parameter is not specified and the file already exists.
     Writer : ``Writer``
         Writer class (DEPRECATED) (default= :class:`Basic`)
     """
+    overwrite = kwargs.pop('overwrite', None)
+    if isinstance(output, six.string_types):
+        if os.path.isfile(output):
```

Comment:
I prefer letting it bubble up from elsewhere, since the error message is clearer (e.g. in case of a directory). But, with the current implementation, a broken symlink gets silently overwritten (which may or may not be what the user wants), so probably `lexists()` is better. The error message is broad enough anyway to cover directories as well.